extensions/bombman/client.mjs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+
+const keyboardMapping = {
+  place: ' ',
+};
+
+/**
+ * This class is the browser/client side of an extension.
+ * One instance is created for each connected player.
+ */
+class Client {
+  /**
+   * Create the client side of the extension.
+   * @constructor
+   * @param {ClientExtensionHelper} helper - An extension helper object for
+   * servicing various functionalities of the extension.
+   */
+  constructor(helper) {
+    this.helper = helper;
+  }
+
+  /**
+   * The initialization function.
+   */
+  async gameStart() {
+    this.helper.inputManager.registerKeydown(this.helper.mapRenderer.getCanvas(), this.placeBomb.bind(this));
+    // TODO: player cool down
+  }
+
+  /**
+   * Callback function of map keydown.
+   * Place a bomb if key is `keyboardMapping.place`.
+   * @param {KeyboardEvent} event - the keydown event
+   */
+  async placeBomb(event) {
+    if (event.key === keyboardMapping.place) {
+      // TODO: reduce latency using the same mechanism of player moving
```

Comment:
There is probably little to no additional latency with this call. They both go through TCP/socket.io.